docs/poc2023q1/edge-scheduler.md

```diff
@@ -0,0 +1,65 @@
+### Edge scheduler
+The edge scheduler monitors the EdgePlacement, Location, and SyncTarget objects and maintains the results of matching.
+
+It starts from the kcp-scheduling-placement controller separated from kcp-dev/kcp, now being refactored towards the goal above.
+
+It works with [this snapshot](https://github.com/kcp-dev/kcp/tree/4506fdc064060b3fe82e1082533f9798b36ba7a5) of kcp.
+
+#### A short demo
+Start a kcp server with [this snapshot](https://github.com/kcp-dev/kcp/tree/4506fdc064060b3fe82e1082533f9798b36ba7a5).
+
+Point `$KUBECONFIG` to the admin kubeconfig for that kcp server.
+
+Workspace `root:edge` is used as the edge service provider workspace.
+```console
+kubectl ws root
+kubectl ws create edge --enter
+```
+
+Create CRDs and expose APIs.
+```console
+kubectl apply -f config/crds/ -f config/exports/
+```
+
+User home workspace is used as the workload management workspace.
+```console
+kubectl ws \~
+```
+
+Bind APIs and create an EdgePlacement CR.
+```console
+kubectl apply -f config/imports/
```

Comment:
For this one we could use the command `kubectl kcp bind export`

Reply:
I have a preference for showing exactly what is going on, as this text currently does.  I would rather see a note that the `kubectl apply` is equivalent to `kubectl kcp bind` than a replacement.

Reply:
There is some interesting thing here. Look like there is some bug with this command. Specifically, this command doesn't like the APIExport's name `edge.kcp.io`. I have two APIExports, where the only difference is their name. One is named `edge.kcp.io`, the other is named `edge`. Only the second can be bound.

I have the text from my terminal:
```
👉 edge-mc $ kc ws root:edge
Current workspace is "root:edge".
👉 edge-mc $ kc get apiexport
NAME          AGE
edge          7m12s
edge.kcp.io   16h
👉 edge-mc $ kc get apiexport edge.kcp.io -oyaml | yq .spec
identity:
  secretRef:
    name: edge.kcp.io
    namespace: kcp-system
latestResourceSchemas:
  - v230126-0a8bf97.edgeplacements.edge.kcp.io
  - v230207-d7d166d.singleplacementslice.edge.kcp.io
👉 edge-mc $ kc get apiexport edge -oyaml | yq .spec
identity:
  secretRef:
    name: edge
    namespace: kcp-system
latestResourceSchemas:
  - v230126-0a8bf97.edgeplacements.edge.kcp.io
  - v230207-d7d166d.singleplacementslice.edge.kcp.io
👉 edge-mc $ kc ws \~
Current workspace is "kvdk2spgmbix".
👉 edge-mc $ kc kcp bind apiexport root:edge:edge.kcp.io
error: fully qualified reference to workspace where APIExport exists is required. The format is `<logical-cluster-name>:<apiexport>` or `<full>:<path>:<to>:<apiexport>`
👉 edge-mc $ kc kcp bind apiexport root:edge:edge
apibinding edge created. Waiting to successfully bind ...
edge created and bound.
👉 edge-mc $ 
```

But I don't know whether I should file this bug. Because I am using the `kubectl-kcp` binary built from the specific commit that edge-mc depends on. Didn't try the latest `kubectl-kcp` binary.

Reply:
I would test with the latest code before opening an issue.

You could add a note of equivalency with the command that works, without getting into the issue of the command that does not work.

Reply:
Thanks for the suggestion @MikeSpreitzer ! Actually can I do the test with the latest code and open an issue if the bug is still there? Wanted to grasp this small opportunity to get more involved with kcp-dev/kcp.

Reply:
Hmm looks there is https://github.com/kcp-dev/kcp/issues/2773 already opened a few days ago.